File: apps/Windows/src/IssuesKit/Catalog.cs
namespace IssuesKit;

/// <summary>
/// A label the project uses, cached in the document so pickers work without network access.
/// </summary>
public sealed record LabelDefinition
{
    /// <summary>The value used in <see cref="Issue.Labels"/>.</summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>A <c>RRGGBB</c> or <c>#RRGGBB</c> colour, or <c>null</c> when the label has none of its own.</summary>
    public string? ColorHex { get; set; }

    public string Description { get; set; } = string.Empty;
}

/// <summary>
/// A milestone the project uses, cached in the document so pickers work without network access.
/// </summary>
public sealed record Milestone
{
    /// <summary>The value used in <see cref="Issue.Milestone"/>.</summary>
    public string Name { get; set; } = string.Empty;

    public DateTimeOffset? DueOn { get; set; }

    public bool IsClosed { get; set; }
}

/// <summary>
/// Someone who can report or be assigned an issue, cached in the document so pickers work
/// without network access.
/// </summary>
public sealed record Person
{
    /// <summary>The tracker handle used in <see cref="Issue.Assignees"/> and <see cref="Issue.ReportedBy"/>.</summary>
    public string Handle { get; set; } = string.Empty;

    public string DisplayName { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;
}

File: apps/Android/issueskit/src/main/kotlin/com/druware/issueskit/Catalog.kt
package com.druware.issueskit

import java.time.Instant

/** A label the project uses, cached in the document so pickers work without network access. */
data class LabelDefinition(
    val name: String = "",
    /** A `RRGGBB` or `#RRGGBB` colour, or `null` when the label has no colour of its own. */
    val colorHex: String? = null,
    val description: String = "",
) {
    val id: String get() = name
}

/** A milestone the project uses, cached in the document so pickers work without network access. */
data class Milestone(
    val name: String = "",
    /** A day-normalized date — see [IssueDate]. */
    val dueOn: Instant? = null,
    val isClosed: Boolean = false,
) {
    val id: String get() = name
}

/**
 * Someone who can report or be assigned an issue, cached in the document so pickers work without
 * network access.
 */
data class Person(
    /** The tracker handle used in [Issue.assignees] and [Issue.reportedBy]. */
    val handle: String = "",
    val displayName: String = "",
    val email: String = "",
) {
    val id: String get() = handle
}
